src/main/java/com/google/devtools/build/lib/analysis/starlark/Args.java

```diff
@@ -101,8 +104,13 @@ public void debugPrint(Printer printer, StarlarkSemantics semantics) {
    */
   public abstract ImmutableSet<Artifact> getDirectoryArtifacts();
 
+  public interface InterruptibleRepoMappingSupplier {
```

Comment:
We have an `InterruptibleSupplier` in `src/main/java/com/google/devtools/build/lib/supplier/`. You can use that.